handlers/api.go

```diff
@@ -3231,6 +3232,71 @@ func insertStats(userData *types.UserWithPremium, machine string, body *map[stri
 	return nil
 }
 
+// ApiWithdrawalCredentialsValidators godoc
+// @Summary Get validator indexes and pubkeys of a withdrawal credential or eth1 address
+// @Tags Validator
+// @Description Returns the validator indexes and pubkeys of a withdrawal credential or eth1 address
+// @Produce json
+// @Param withdrawalCredentialsOrEth1address path string true "Provide a withdrawal credential or an eth1 address with an optional 0x prefix"
+// @Param  limit query int false "Limit the number of results, maximum: 200" default(10)
+// @Param offset query int false "Offset the number of results" default(0)
+// @Success 200 {object} types.ApiResponse{data=[]types.ApiWithdrawalCredentialsResponse}
+// @Failure 400 {object} types.ApiResponse
+// @Router /api/v1/validator/withdrawalCredentials/{withdrawalCredentialsOrEth1address} [get]
+func ApiWithdrawalCredentialsValidators(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("Content-Type", "application/json")
+
+	vars := mux.Vars(r)
+	q := r.URL.Query()
+
+	credentialsOrAddressString := vars["withdrawalCredentialsOrEth1address"]
+	credentialsOrAddressString = strings.ToLower(credentialsOrAddressString)
+	credentialsOrAddress := common.FromHex(credentialsOrAddressString)
+
+	credentials := utils.AddressToWithdrawalCredentials(credentialsOrAddress)
```

Comment:
There is no sanity check. Call the function with `withdrawalCredentialsOrEth1address` set to `123` and the function will just accept that, query the db with that (=meaningless traffic) and return something like
```
{"status":"OK","data":[]}
``` 
For your convenience: http://localhost:8081/api/v1/validator/withdrawalCredentials/123

Take a look at for example https://beaconcha.in/api/v1/validator/eth1/324234234

I suggest that invalid parameters are checked and if the parameters are invalid, an error with a good description is returned. Something like
_"invalid parameter provided, neither withdrawal credentials nor eth1 address"_
(don't copy&paste this please, this is just an example I did not ponder too much)

Furthermore I think its weird to have a function called `AddressToWithdrawalCredentials` that accepts pretty much anything. Based on its name, I would expect an error if I pass `123`. Instead it returns `123` and I have no information if I can work with that or not.

My suggestion would be to remove `AddressToWithdrawalCredentials` and do everything (including sanity checks, error handling and "casting") within `ApiWithdrawalCredentialsValidators` directly.
Alternatively, have something SIMILAR to `AddressToWithdrawalCredentials` that returns withdrawalCredentials and an error based on an input that is not called `address` but `withdrawalCredentialsOrAddress`. Make the sanity checks there and return `nil` as error if the return value are valid withdrawalCredentials or return `nil` as credentials and an error based on your sanity checks. I cannot come up with a good function name for that though :smile: 

Thoughts?

Reply:
As discussed I rewrote WithdrawalCredentialsToAddress and AddressToWithdrawalCredentials to also return errors.

I also noticed that all the "0x" regexp checks were wrong so I corrected them.